test/presentation/bloc/bloc_genre_list_test.dart

```diff
@@ -20,9 +24,10 @@ void main() async {
     late GenreBloc _blocGenreList;
     setUp(() {
       _blocGenreList = GenreBloc(
-          genreListUseCase: GenreUseCase(
-              genreRepository:
-                  GenreRemoteRepository(apiService: ApiService())));
+          genreUseCase: GenreUseCase(
```

Comment:
Why don't you mock your `GenreUseCase` directly instead of instantiating your whoooooooooole dependency tree and end up mocking the first dependency?
Please refactor this test and the rest of tests similar to this one